lucene/codecs/src/java/org/apache/lucene/codecs/uniformsplit/IndexDictionary.java

```diff
@@ -97,31 +100,47 @@
      * Otherwise {@code -1} if there is no floor block key because the searched
      * term precedes alphabetically the first block key of the dictionary.
      */
-    long seekBlock(BytesRef term);
+    long seekBlock(BytesRef term) throws IOException;
 
     /**
      * Returns the next block key and positions the browser at this key.
      * A key is a prefix of a term in the dictionary.
      * If seekBlock was just called then this is the current block key.
      */
-    BytesRef nextKey();
+    BytesRef nextKey() throws IOException;
 
     /**
      * Returns the next key without advancing.
      * Only call this after {@link #nextKey()} returns a non-null result.
      */
-    BytesRef peekKey();
+    BytesRef peekKey() throws IOException;
 
     /**
      * Returns the number of characters of this block's key that is in common with all terms in this block.
      * Only call this after {@link #nextKey()} returns a non-null result.
      */
-    int getBlockPrefixLen();
+    int getBlockPrefixLen() throws IOException;
 
     /**
      * Returns the block file pointer associated with the key returned.
      * Only call this after {@link #nextKey()} returns a non-null result.
      */
-    long getBlockFilePointer();
+    long getBlockFilePointer() throws IOException;
+  }
+
+  /**
+   * Supplier for a new stateful {@link Browser} created on the immutable {@link IndexDictionary}.
+   * <p>
+   * The immutable {@link IndexDictionary} is lazy loaded thread safely. This lazy loading allows
+   * us to load it only when {@link org.apache.lucene.index.TermsEnum#seekCeil} or
+   * {@link org.apache.lucene.index.TermsEnum#seekExact} are called (it is not loaded for a direct
+   * all-terms enumeration).
+   */
+  interface BrowserSupplier extends Accountable {
```

Comment:
See `org.apache.lucene.util.IOSupplier` which this basically duplicates.  I'm not sure we need to mandate Accountable; it could be optionally implemented by an implementation?

Reply:
Thanks. I looked externally for such supplier but I didn't think of looking in Lucene util...
Accountable is leveraged in UniformSplitTerms.getDictionaryRamBytesUsed() when computing the heap space. We could indeed put there an instanceof check and optionally cast to Accountable. But I'd prefer to mandate Accountable so that any implementation doesn't forget to provide it.
What about keeping BrowserSupplier to join IOSupplier and Accountable (without additional interface contract)?

Reply:
> What about keeping BrowserSupplier to join IOSupplier and Accountable (without additional interface contract)?

+1